integration/spark/src/main/scala/org/apache/spark/sql/execution/command/index/CarbonRefreshIndexCommand.scala

```diff
@@ -71,30 +80,81 @@ case class CarbonRefreshIndexCommand(
   private def refreshIndex(
       sparkSession: SparkSession,
       parentTable: CarbonTable,
-      indexOp: Optional[DataMapSchema]): Unit = {
-    val schema = indexOp.get
+      indexMetaData: IndexMetadata): Unit = {
+    var indexInfo: util.Map[String, String] = new util.HashMap[String, String]()
+    val allIndexesIterator = indexMetaData.getIndexesMap.entrySet().iterator()
+    breakable {
+      while (allIndexesIterator.hasNext) {
+        val currentIndex = allIndexesIterator.next()
+        if (!currentIndex.getKey.equalsIgnoreCase(CarbonIndexProvider.SI.getIndexProviderName)) {
```

Comment:
If we are excluding SI everywhere, It is better to have separate map for each provider in `IndexMetadata`. access only Non SI maps.

or you can directly lookup non SI providers

Reply:
handled